sources/source-Send_to_MQTT.module

```diff
@@ -0,0 +1,89 @@
+<?php
+/*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
+ * Developer Notes:
+ *
+ * Version History
+ *     2020-10-14  Initial creation by hartmann1970@gmx.de
+ *     2020-10-15 added user, password, clientID by hartmann1970@gmx.de
+ *
+ *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***/
+
+class Send_to_MQTT extends superfecta_base {
+
+    public $description = "This source will send CID number and name to an MQTT topic for external processing. The MQTT topic is added with DID and payload is JSON format. To use this feature, mosqitto has to be installed.";
+    public $version_requirement = "2.11";
+    public $source_param = array(
+        'MQTT_broker' => array(
+            'description' => 'Specify FQDN or IP of MQTT broker. Use the format \'this.is.org\' or \'192.168.123.045\'',
+            'type' => 'text',
+            'default' => "10.0.0.10"
+        ),
+        'MQTT_port' => array(
+            'description' => 'Specify the port of the MQTT broker. Use the format \'1883\'',
+            'type' => 'number',
+            'default' => "1883"
+        ),
+        'MQTT_topic' => array(
+            'description' => 'Specify the topic to send CID/CNAM data to. Use the format \'this/is/my/topic\' without trailing slash.',
+            'type' => 'text',
+            'default' => 'freepbx/calls'
+        ),
+        'MQTT_user' => array(
+            'description' => 'Specify the user. Neither spaces nor special characters allowed.',
+            'type' => 'text',
+            'default' => ''
+        ),
+        'MQTT_password' => array(
+            'description' => 'Specify the password. Neither spaces nor special characters allowed.',
+            'type' => 'text',
+            'default' => ''
+        ),
+        'MQTT_clientID' => array(
+            'description' => 'Specify the ClientID. Neither spaces nor special characters allowed.',
+            'type' => 'text',
+            'default' => 'freePBX'
+        ),
+        'Default_DID' => array(
+            'description' => 'In cases where DID is unknown, substitute this string in its place.',
+            'type' => 'text',
+            'default' => 'Unknown'
+        )
+    );
+
+    function post_processing($cache_found, $winning_source, $first_caller_id, $run_param, $thenumber) {
+
+        if (($run_param['MQTT_broker'] != '') && ($run_param['MQTT_topic'] != '') && ($first_caller_id != '')) {
+            $port = ' -p '. $run_param['MQTT_port'];
+            $broker = ' -h '. $run_param['MQTT_broker'];
+            $user = "";
+            if($run_param['MQTT_user'] != ''){
+                $user = ' -u '. $run_param['MQTT_user'];
+            };
+            $password = "";
+            if($run_param['MQTT_user'] != ''){
```

Comment:
Hi @udo1toni  - I can't test, just scanning thru the code. Is this line correct, check to see if user is "" and then set password?

Reply:
copy/paste error :) clearly it should be MQTT_password...